### DISCUSSION TITLE
PageHeader sometimes state update error

### DISCUSSION THREAD

**Discussion by wuifdesign** [ant-design/ant-design#33296] [Q&A]
As `PageHeader` uses `ResizeObserver` and this is fired on resize, somtime the onResize is triggered with `{width: 0, height: 0, offsetWidth: 0, offsetHeight: 0}`, but element was already removed. 

```
const onResize = ({ width }: { width: number }) => {
  updateCompact(width < 768);
};
```

This results in following error:

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.
```

I think this happens because of this code inside the `SingleObserver` (rc-resize-observer):

```
 if (onResize) {
        // defer the callback but not defer to next frame
        Promise.resolve().then(() => {
          onResize(sizeInfo, target);
        });
      }
```

any workaround for this? 

also i think it may have to do something with using the `Suspense` element. for lazy loading elements.

**Comment by Kamahl19** [comment]
@wuifdesign I believe this is a bug and should be reported as such. Either in this repo or in https://github.com/react-component/resize-observer

**Reply by wuifdesign** [reply]
done ;) https://github.com/ant-design/ant-design/issues/33513
